packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js

```diff
@@ -285,14 +295,16 @@ class EnumerationWidget extends React.Component {
 			return;
 		}
 
-		if (!this.valueAlreadyExist(value.value)) {
-			this.setState({
-				displayMode: 'DISPLAY_MODE_DEFAULT',
-				items: this.state.items.concat([{
-					values: [value.value],
-				}]),
-			}, this.setFormData.bind(this));
-			this.updateHeaderInputDisabled('');
+		if (!this.callActionHandler(ENUMERATION_ADD_ACTION, value.value)) {
```

Comment:
we can't know if the values are not all in the list & we want the same behaviour if it's in the list or not